java/spring/security/injection/tainted-html-string.yaml

```diff
@@ -32,46 +32,74 @@ rules:
-  - patterns:
+  - label: INPUT
+    patterns:
     - pattern-either:
       - pattern-inside: |
-          $METHODNAME(..., @$REQ(...) $TYPE 
-          $SOURCE,...) {
+          $METHODNAME(..., @$REQ(...) $TYPE $SOURCE,...) {
             ...
           }
       - pattern-inside: |
-          $METHODNAME(..., @$REQ $TYPE 
-          $SOURCE,...) {
+          $METHODNAME(..., @$REQ $TYPE $SOURCE,...) {
             ...
           }
     - metavariable-regex:
         metavariable: $REQ
         regex: (RequestBody|PathVariable|RequestParam)
-    - pattern: $SOURCE
+    - focus-metavariable: $SOURCE
+  - label: CONCAT
+    by-side-effect: true
+    requires: INPUT
+    patterns:
+    - pattern-either:
+      - pattern: |
+          "$HTMLSTR" + ...
+      - pattern: |
+          "$HTMLSTR".concat(...)
+      - patterns:
+        - pattern-inside: |
+            StringBuilder $SB = new StringBuilder("$HTMLSTR");
+            ...
+        - pattern: $SB.append(...)
+      - patterns:
+        - pattern-inside: |
+            $VAR = "$HTMLSTR";
+            ...
+        - pattern: $VAR += ...
+      - pattern: String.format("$HTMLSTR", ...)
+      - patterns:
+        - pattern-inside: |
+            String $VAR = "$HTMLSTR";
+            ...
+        - pattern: String.format($VAR, ...)
+    - metavariable-regex:
+        metavariable: $HTMLSTR
+        regex: ^<\w+
+  pattern-propagators:
+  - pattern: (StringBuilder $SB).append($...TAINTED)
+    from: $...TAINTED
+    to: $SB
+  - pattern: $VAR += $...TAINTED
+    from: $...TAINTED
+    to: $VAR
   pattern-sinks:
-  - patterns:
-    - patterns:
-      - pattern-either:
-        - pattern: |
-            "$HTMLSTR" + ...
-        - pattern: |
-            "$HTMLSTR".concat(...)
-        - patterns:
-          - pattern-inside: |
-              StringBuilder $SB = new StringBuilder("$HTMLSTR");
-              ...
-          - pattern: $SB.append(...)
-        - patterns:
-          - pattern-inside: |
-              $VAR = "$HTMLSTR";
-              ...
-          - pattern: $VAR += ...
-        - pattern: String.format("$HTMLSTR", ...)
-        - patterns:
-          - pattern-inside: |
-              String $VAR = "$HTMLSTR";
-              ...
-          - pattern: String.format($VAR, ...)
-      - metavariable-regex:
-          metavariable: $HTMLSTR
-          regex: ^<\w+
+  - requires: CONCAT
+    patterns:
+    - pattern-either:
+      - pattern: new ResponseEntity<>($PAYLOAD, ...)
+      - pattern: new ResponseEntity<$ERROR>($PAYLOAD, ...)
+      - pattern: ResponseEntity. ... .body($PAYLOAD)
+      - patterns:
+        - pattern-either:
+          - pattern: ResponseEntity.$RESPFUNC($PAYLOAD). ... .$FUNC(...)
+          - pattern: ResponseEntity.$RESPFUNC($PAYLOAD)
+        - metavariable-regex:
+            metavariable: $RESPFUNC
+            regex: ^(ok|of)$
+    - focus-metavariable: $PAYLOAD
+  pattern-sanitizers:
+  - pattern-either:
+    - pattern: Encode.forHtml(...)
```

Comment:
may want to turn these into $METAVARIABLE to make it generic but overall should be fine.

Reply:
will make a note, I have these sanitizers for other XSS rules so will try to fix them all at once later 👍